SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp

```diff
@@ -64,6 +73,42 @@ void USpatialGDKSettings::PostEditChangeProperty(FPropertyChangedEvent& Property
 			FModuleManager::LoadModuleChecked<ISettingsModule>("Settings").ShowViewer("Project", "SpatialGDKEditor", "Editor Settings");
 		}
 	}
+
+	if (PropertyName == GET_MEMBER_NAME_CHECKED(USpatialGDKSettings, ActorGroups))
+	{
+		ValidateOffloadingSettings();
+	}
+
+	if (PropertyName == GET_MEMBER_NAME_CHECKED(USpatialGDKSettings, WorkerTypes))
+	{
+		ValidateOffloadingSettings();
+	}
+
 	Super::PostEditChangeProperty(PropertyChangedEvent);
 }
+
+void USpatialGDKSettings::PostEditChangeChainProperty(FPropertyChangedChainEvent& PropertyChangedEvent)
```

Comment:
Is there a handler which gets called on undo operations? The issue is that if you delete an actor group and then undo the change, the worker associations are only updated with the remove operation and the group is not restored. Overall, I think there are quite a lot of inconsistencies that could arise by inconsistent state of the internal data structures which track worker types, actor groups and worker association.